superset/utils/date_parser.py

```diff
@@ -322,10 +323,12 @@ def eval(self) -> datetime:
             dttm = dttm.replace(
                 month=1, day=1, hour=0, minute=0, second=0, microsecond=0
             )
+        if unit == "quarter":
+            dttm = datetime(dttm.year, 3 * pd.Timestamp(dttm).quarter - 2, 1)
         elif unit == "month":
             dttm = dttm.replace(day=1, hour=0, minute=0, second=0, microsecond=0)
         elif unit == "week":
-            dttm = dttm - relativedelta(days=dttm.weekday())
+            dttm = -relativedelta(days=dttm.weekday())
```

Comment:
Is this intentional? Or did you mean `ddtm -= relativedelta(...)`?

Reply:
Oops.